keanu-project/src/main/java/io/improbable/keanu/algorithms/variational/optimizer/gradient/GradientOptimizer.java

```diff
@@ -40,6 +41,15 @@
     }
 
     public static GradientOptimizer of(BayesianNetwork bayesNet) {
+        long numIntegerVertices = bayesNet.getDiscreteLatentVertices().stream()
+            .filter(v -> v instanceof IntegerVertex)
+            .count();
+
+        if (numIntegerVertices != 0) {
+            throw new UnsupportedOperationException("Gradient Optimization unsupported on Networks containing " +
+                "Discrete Latents");
```

Comment:
is it worth mentioning in this error message how many offending discrete vertices there are?

Reply:
I had thought about that - but given we don't really give any indication of *which* vertices are causing the issue I wasn't sure it was that useful (ie you have to look through your code to figure out what you're doing wrong anyway).  I'll add though as a bit of extra information can't hurt!  When we have Vertex labelling, we'll be able to improve this sort of error further.

Reply:
ye good point, vertex lablels would be the only useful information maybe